typescript/infra/scripts/funding/fund-keys-from-deployer.ts

```diff
@@ -407,25 +393,23 @@ class ContextFunder {
   }
 
   private getChainKeys() {
-    const entries = AllChains.map((c) => {
-      return [c, []];
-    });
-    const chainKeys: ChainMap<BaseCloudAgentKey[]> =
-      Object.fromEntries(entries);
+    const chainKeys: ChainMap<BaseCloudAgentKey[]> = Object.fromEntries(
+      // init with empty arrays
+      AllChains.map((c) => [c, []]),
+    );
     for (const role of this.rolesToFund) {
       const keys = this.getKeysWithRole(role);
-      for (const chain of this.chains) {
-        if (role === KEY_ROLE_ENUM.Relayer) {
-          // Relayer keys should not be funded on the origin chain
-          for (const remote of this.chains) {
-            chainKeys[remote] = chainKeys[remote].concat(
-              keys.filter(
-                (_) => _.chainName !== remote && _.chainName === chain,
-              ),
-            );
+      for (const key of keys) {
+        if (key.role == Role.Relayer || key.role == Role.Kathy) {
```

Comment:
Do we need this condition at all? I think we cover this in knowing that `rolesToFund` is limited to only relayer or kathy in the `validRoles` check below